### PR TITLE
biflags: new recipe

### PR DESCRIPTION
**Pull request by robomics**
Specify library name and version:  **bitflags/1.5.0**

Contribute recipe for [m-peko/bitflags](https://github.com/m-peko/bitflags), a header-only library for easily managing set of auto-generated type-safe flags.

I am not the author of this library, but some of my projects depend on it.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/bitflags/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    def package_id(self):
+        self.info.clear()
+
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "bitflags")
+        self.cpp_info.set_property("cmake_target_name", "bitflags::bitflags")
+
+        # TODO: to remove in conan v2 once cmake_find_package* generators removed
+        self.cpp_info.names["cmake_find_package"] = "bitflags"
+        self.cpp_info.names["cmake_find_package_multi"] = "bitflags"
```
Comment (SpaceIm) [inline recipes/bitflags/all/conanfile.py:60]:
```suggestion
```

not needed

File: recipes/bitflags/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        get(self, **self.conan_data["sources"][self.version], strip_root=True,
+            destination=self.source_folder)
+
+    def package(self):
+        copy(self, pattern="LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        copy(self, pattern="*", src=os.path.join(self.source_folder, "include"), dst=os.path.join(self.package_folder, "include"))
+
+    def package_id(self):
+        self.info.clear()
+
```
Comment (SpaceIm) [inline recipes/bitflags/all/conanfile.py:53]:
```suggestion
```

File: recipes/bitflags/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    def _minimum_compilers_version(self):
+        return {"apple-clang": "5", "clang": "5", "gcc": "7", "Visual Studio": "14"}
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 11
+
+    def layout(self):
+        basic_layout(self)
+
```
Comment (SpaceIm) [inline recipes/bitflags/all/conanfile.py:33]:
```suggestion

    def package_id(self):
        self.info.clear()

```

File: recipes/bitflags/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    def validate(self):
+        if self.settings.get_safe("compiler.cppstd"):
+            check_min_cppstd(self, self._minimum_cpp_standard)
+        try:
+            if Version(self.settings.compiler.version) < self._minimum_compilers_version[str(self.settings.compiler)]:
+                raise ConanInvalidConfiguration(f"{self.name} requires a compiler that supports C++{self._minimum_cpp_standard}. {self.settings.compiler}, {self.settings.compiler.version}")
```
Comment (SpaceIm) [inline recipes/bitflags/all/conanfile.py:39]:
```suggestion
                raise ConanInvalidConfiguration(f"{self.ref} requires a compiler that supports C++{self._minimum_cpp_standard}.)
```

File: recipes/bitflags/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    def package(self):
+        copy(self, pattern="LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        copy(self, pattern="*", src=os.path.join(self.source_folder, "include"), dst=os.path.join(self.package_folder, "include"))
+
+    def package_id(self):
+        self.info.clear()
+
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "bitflags")
+        self.cpp_info.set_property("cmake_target_name", "bitflags::bitflags")
```
Comment (SpaceIm) [inline recipes/bitflags/all/conanfile.py:56]:
```suggestion
        self.cpp_info.set_property("cmake_target_name", "bitflags::bitflags")
        self.cpp_info.bindirs = []
        self.cpp_info.libdirs = []
        self.cpp_info.resdirs = []
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by toge — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**